canal/rows.go

```diff
@@ -16,6 +18,8 @@ const (
 
 // RowsEvent is the event for row replication.
 type RowsEvent struct {
+	// the position in this event
+	CurPos mysql.Position
```

Comment:
Can your application read the last position from `replication.EventHeader` or use `LogPos-EventSize`? So we don't need change this library.

Reply:
就我的理解 header的LogPos 是next position. 由于log rotate , 那么要获取当前event对应的position和logName 需要记录rotate前的logName 我觉得这个既然在 runSyncBinlog的主循环能拿到 为什么不直接记录到event呢?

Reply:
There're many fake event so that last event's end position is not next event's begin position. I'm not sure if they are totally checked. For example, heartbeat event in MySQL and MariaDB GTID list event

Reply:
所以意思是我并不能准确拿到当前event对应的position是吗?

Reply:
I think you can use `LogPos-EventSize`

considering

RowEvent1 EventSize=10, LogPos=110,
SomeFakeEvent EventSize=undetermined, LogPos=undetermined,
RowEvent2 EventSize=10, LogPos=120,

Now RowEvent2 has a start position of 110. But its previous event is SomeFakeEvent, the LogPos may not be 110 or 0.

To detect SomeFakeEvent we have found some rules, like LogPos=0, LOG_EVENT_ARTIFICIAL_F, but I don't have confidence that we list them totally.

Reply:
好的 我大概明白了 LogPos-EventSize 应该也能满足需求 pr我close掉了